Source/SwiftLintFramework/Rules/MultilineFunctionChainsRule.swift

```diff
@@ -0,0 +1,221 @@
+
+    private func callHasLeadingNewline(file: File, callRange: NSRange) -> Bool {
+        guard
+            let range = file.contents.bridge().byteRangeToNSRange(start: callRange.location, length: callRange.length),
+            case let regex = type(of: self).newlineWhitespaceDotRegex,
+            regex.firstMatch(in: file.contents, options: [], range: range) != nil else {
+                return false
+        }
+        return true
+    }
+
+    private func callRanges(file: File,
+                            kind: SwiftExpressionKind,
+                            dictionary: [String: SourceKitRepresentable],
+                            parentCallName: String? = nil) -> [NSRange] {
+        guard
+            kind == .call,
+            case let contents = file.contents.bridge(),
+            let offset = dictionary.nameOffset,
+            let length = dictionary.nameLength,
+            let name = contents.substringWithByteRange(start: offset, length: length) else {
+                return []
+        }
+
+        let subcalls = dictionary.subcalls
+
+        if subcalls.isEmpty, let parentCallName = parentCallName, parentCallName.starts(with: name) {
+            return [NSRange(location: offset, length: length)]
+        }
+
+        return subcalls.flatMap { call -> [NSRange] in
+            guard let range = callRange(file: file, call: call, parentName: name, parentNameOffset: offset) else {
+                return []
+            }
+
+            return [range] + callRanges(file: file, kind: .call, dictionary: call, parentCallName: name)
+        }
+    }
+
+    private func callRange(file: File,
+                           call: [String: SourceKitRepresentable],
+                           parentName: String,
+                           parentNameOffset: Int) -> NSRange? {
+        guard
+            case let contents = file.contents.bridge(),
+            let nameOffset = call.nameOffset,
+            parentNameOffset == nameOffset,
+            let nameLength = call.nameLength,
+            let bodyOffset = call.bodyOffset,
+            let bodyLength = call.bodyLength,
+            let name = contents.substringWithByteRange(start: nameOffset, length: nameLength),
+            parentName.starts(with: name) else {
+                return nil
+        }
+
+        let linkOffset = nameOffset + nameLength
+        let linkLength = parentName.bridge().length - nameLength
+        let offsetDifference = bodyOffset - linkOffset
+
+        return NSRange(location: linkOffset + offsetDifference + bodyLength,
+                       length: linkLength - bodyLength - offsetDifference)
+    }
+
+}
+
+fileprivate extension Dictionary where Key: ExpressibleByStringLiteral {
+    var subcalls: [[String: SourceKitRepresentable]] {
+        return substructure.compactMap { dictionary -> [String: SourceKitRepresentable]? in
+            guard case .call? = dictionary.kind.flatMap({ SwiftExpressionKind(rawValue: $0) }) else {
```

Comment:
small nitpick: you can use `dictionary.kind.flatMap(SwiftExpressionKind.init) == .call`